src/utils/auth.js

```diff
@@ -25,9 +38,27 @@ function backendAuthorization(req, res, next) {
   else return passport.authenticate('keycloak', { session: false })(req, res, next);
 }
 
+/**
+ * Middleware for verifying the access token on a Subscription notification
+ */
 function subscriptionAuthorization(req, res, next) {
-  // TODO: implement this function to check the Subscription token matched the request
-  return next();
+  const token = getToken(req);
+  if (token === 'admin') return next();
+  else if (token) {
+    const subscriptionId = token.split(':')[0];
+    if (subscriptionId) {
+      const subscription = db.select(SUBSCRIPTIONS, s => s.id === subscriptionId);
+      if (subscription.length) {
+        // Get the authorization token from the subscription
+        const authorizationHeader = subscription[0].channel.header[0];
+        const authToken = authorizationHeader.split('Authorization: Bearer ')[1];
+        if (token === authToken) return next();
+      }
+    }
+  }
+
+  res.sendStatus(StatusCodes.UNAUTHORIZED);
+  return;
```

Comment:
Can this `return` statement be removed? Function is going to end anyway right?

Reply:
Yea you're right. Originally it wasn't at the end so it was needed but then I refactored the branching. I will remove it